Dockerfile

```diff
@@ -53,133 +91,220 @@ RUN --mount=type=cache,target=/tmp/.gradle/ \
+RUN . $NVM_DIR/nvm.sh && nvm install "${NODEJS_VERSION}"
+RUN . $NVM_DIR/nvm.sh \
+    && npm install --global npm@$NPM_VERSION bower@$BOWER_VERSION yarn@$YARN_VERSION
+COPY docker/nodejs.sh /etc/profile.d
+
+#------------------------------------------------------------------------
+# GOLANG - Build as a separate component
+FROM build AS gobuild
+
+ARG GO_DEP_VERSION=0.5.4
+ARG GO_VERSION=1.17.3
+ENV GOPATH=/opt/go
+RUN curl -L https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz | tar -C /opt -xz
+ENV PATH=/opt/go/bin:$PATH
+RUN go version
+RUN curl -ksS https://raw.githubusercontent.com/golang/dep/v$GO_DEP_VERSION/install.sh | bash
+RUN echo "add_local_path /opt/go/bin:\$PATH" > /etc/profile.d/go.sh
+
+#------------------------------------------------------------------------
+# HASKELL STACK
+FROM build AS haskellbuild
+
+ARG HASKELL_STACK_VERSION=2.1.3
+RUN curl -ksS https://raw.githubusercontent.com/commercialhaskell/stack/v$HASKELL_STACK_VERSION/etc/scripts/get-stack.sh | bash -s -- -d /usr/bin
+
+#------------------------------------------------------------------------
+# REPO / ANDROID SDK
+FROM build AS androidbuild
+
+ARG ANDROID_SDK_VERSION=6858069
+ENV ANDROID_HOME=/opt/android-sdk
+
+RUN curl -ksS https://storage.googleapis.com/git-repo-downloads/repo > /usr/bin/repo \
+    && chmod a+x /usr/bin/repo
+
+RUN curl -Os https://dl.google.com/android/repository/commandlinetools-linux-${ANDROID_SDK_VERSION}_latest.zip \
+    && unzip -q commandlinetools-linux-${ANDROID_SDK_VERSION}_latest.zip -d $ANDROID_HOME \
+    && rm commandlinetools-linux-${ANDROID_SDK_VERSION}_latest.zip \
+    && PROXY_HOST_AND_PORT=${https_proxy#*://} \
+    && if [ -n "$PROXY_HOST_AND_PORT" ]; then \
+       # While sdkmanager uses HTTPS by default, the proxy type is still called "http".
+       SDK_MANAGER_PROXY_OPTIONS="--proxy=http --proxy_host=${PROXY_HOST_AND_PORT%:*} --proxy_port=${PROXY_HOST_AND_PORT##*:}"; \
+    fi \
+    && yes | $ANDROID_HOME/cmdline-tools/bin/sdkmanager $SDK_MANAGER_PROXY_OPTIONS \
+       --sdk_root=$ANDROID_HOME "platform-tools" "cmdline-tools;latest"
+COPY docker/android.sh /etc/profile.d
+
+#------------------------------------------------------------------------
+# SCANCODE - Build ScanCode as a separate component
+FROM pythonbuild AS scancodebuild
+
+ARG SCANCODE_VERSION="3.2.1rc2"
+ENV SCANCODE_URL "https://github.com/nexB/scancode-toolkit/archive/refs/tags/"
+ENV PATH=/opt/python/shims:$PATH
+
+RUN mkdir -p /opt/scancode \
+    && curl -ksSL ${SCANCODE_URL}/v${SCANCODE_VERSION}.tar.gz | tar -C /opt/scancode -xz --strip-components=1 \
+    && cd /opt/scancode \
+    && PYTHON_EXE=python3 ./configure \
+    # Clean up unneeded installed binaries
+    && rm -rf /opt/scancode/thirdparty
+
+#------------------------------------------------------------------------
+# Main container
+FROM eclipse-temurin:11-jre
+
+# ORT
+COPY --from=ortbuild /opt/ort /opt/ort
```

Comment:
This is the thing, that gets rebuilt nearly every time? Why not do that last so that caching of layers can be leveraged more?

Reply:
I was wondering the same. Looks like because we are applying a sed on runtime over gradle that invalidate the cache, and worst, everytme increase the cache build.
When developing this one sometime i need clean up >100Gb of caches on docker :-/